lib/fluent/plugin/buffer.rb

```diff
@@ -271,6 +312,18 @@ def enqueue_chunk(metadata)
         nil
       end
 
+      def enqueue_unstaged_chunk(chunk)
+        synchronize do
```

Comment:
This method is called in `synchronize` in `write`.
This `synchronize` is needed?